main.c

```diff
@@ -117,11 +121,17 @@ static int min(int a, int b) {
 	return (a < b) ? a : b;
 }
 
+static int max(int a, int b) {
+	return (a > b) ? a : b;
+}
+
 void seat_get_box(struct slurp_seat *seat, struct slurp_box *result) {
-	result->x = min(seat->pressed_x, seat->x);
-	result->y = min(seat->pressed_y, seat->y);
-	result->width = abs(seat->x - seat->pressed_x);
-	result->height = abs(seat->y - seat->pressed_y);
+	int32_t anchor_x = max(0, seat->anchor_x);
```

Comment:
If you release the selection while out of bound - because you want to start @0,0 for example - you don't want to select the negative part of your screen.